instrumentation/servlet/request-2.3/src/main/java/io/opentelemetry/auto/instrumentation/servlet/v2_3/Servlet2Advice.java

```diff
@@ -90,40 +55,17 @@ public static void stopSpan(
       @Advice.Argument(1) final ServletResponse response,
       @Advice.Enter final SpanWithScope spanWithScope,
       @Advice.Thrown final Throwable throwable) {
-    // Set user.principal regardless of who created this span.
-    final Object spanAttr = request.getAttribute(SPAN_ATTRIBUTE);
-    if (spanAttr instanceof Span && request instanceof HttpServletRequest) {
-      final Principal principal = ((HttpServletRequest) request).getUserPrincipal();
-      if (principal != null) {
-        ((Span) spanAttr).setAttribute(MoreTags.USER_NAME, principal.getName());
-      }
-    }
+    if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
+      TRACER.setPrincipal((HttpServletRequest) request);
```

Comment:
I think it's incorrect that we are not setting "user.principal" on the current span here, but it's not related to this PR (which just preserves existing behavior), so I opened a separate issue to track #470